cobbler/modules/managers/isc.py

```diff
@@ -381,40 +364,37 @@ def write_v6_config(self):
             self.logger.info("generating %s" % self.settings_file_v6)
         self.templar.render(template_data, metadata, self.settings_file_v6)
 
-    def restart_dhcp(self, service_name):
+    def restart_dhcp(self, service_name: str) -> int:
         """
         This syncs the dhcp server with it's new config files.
         Basically this restarts the service to apply the changes.
 
-        :param service_name: TODO
+        :param service_name: The name of the DHCP service.
         """
-        # TODO: Reuse the utils method for service restarts
-        return_code_service_restart = 0
-        return_code_service_restart = utils.subprocess_call("{} -t -q".format(service_name), shell=True)
+        return_code_service_restart = utils.subprocess_call("{} -t -q".format(service_name), shell=False)
         if return_code_service_restart != 0:
             self.logger.error("Testing config - {} -t failed".format(service_name))
-        service_restart = "service {} restart".format(service_name)
-        return_code_service_restart = utils.subprocess_call(service_restart, shell=True)
+        return_code_service_restart = utils.service_restart(service_name)
         if return_code_service_restart != 0:
             self.logger.error("{} service failed".format(service_name))
         return return_code_service_restart
 
     def write_configs(self):
-        if self.settings.manage_dhcp_v4 != "0":
+        if self.settings.manage_dhcp_v4:
```

Comment:
You are changing the behavior here in case this setting is `"0"`, as:

```python
>>> bool("0")
True
```

Is this intended?

Reply:
Yes. We have now through the setting a real boolean which is either `True` or `False`, so yes this change is intended.